lib/rubocop/config_obsoletion.rb

```diff
@@ -4,298 +4,102 @@ module RuboCop
-          Style/TrivialAccessors
-        ],
-        parameters: 'Whitelist',
-        alternative: '`Whitelist` has been renamed to `AllowedMethods`.'
-      },
-      {
-        cops: 'Style/IpAddresses',
-        parameters: 'Whitelist',
-        alternative: '`Whitelist` has been renamed to `AllowedAddresses`.'
-      },
-      {
-        cops: 'Naming/HeredocDelimiterNaming',
-        parameters: 'Blacklist',
-        alternative: '`Blacklist` has been renamed to `ForbiddenDelimiters`.'
-      },
-      {
-        cops: 'Naming/PredicateName',
-        parameters: 'NamePrefixBlacklist',
-        alternative: '`NamePrefixBlacklist` has been renamed to ' \
-                     '`ForbiddenPrefixes`.'
-      },
-      {
-        cops: 'Naming/PredicateName',
-        parameters: 'NameWhitelist',
-        alternative: '`NameWhitelist` has been renamed to ' \
-                     '`AllowedMethods`.'
-      },
-      {
-        cops: %w[Metrics/BlockLength Metrics/MethodLength],
-        parameters: 'ExcludedMethods',
-        alternative: '`ExcludedMethods` has been renamed to `IgnoredMethods`.',
-        severity: :warning
-      }
-    ].freeze
-
-    OBSOLETE_ENFORCED_STYLES = [
-      {
-        cop: 'Layout/IndentationConsistency',
-        parameter: 'EnforcedStyle',
-        enforced_style: 'rails',
-        alternative: '`EnforcedStyle: rails` has been renamed to ' \
-                     '`EnforcedStyle: indented_internal_methods`'
-      }
-    ].freeze
-
-    attr_reader :warnings
+    # Can be extended by extension libraries to add their own obsoletions
+    self.files = [DEFAULT_RULES_FILE]
 
     def initialize(config)
       @config = config
+      @rules = load_rules
       @warnings = []
     end
 
-    def reject_obsolete_cops_and_parameters
-      messages = [obsolete_cops, obsolete_parameters,
-                  obsolete_enforced_style].flatten.compact
+    def reject_obsolete!
+      messages = obsoletions.flatten.compact
       return if messages.empty?
 
       raise ValidationError, messages.join("\n")
     end
 
     private
 
-    def obsolete_cops
-      OBSOLETE_COPS.map do |cop_name, message|
-        next unless @config.key?(cop_name) ||
-                    @config.key?(Cop::Badge.parse(cop_name).cop_name)
-
-        message + "\n(obsolete configuration found in " \
-                  "#{smart_loaded_path}, please update it)"
-      end
-    end
-
-    def obsolete_enforced_style
-      OBSOLETE_ENFORCED_STYLES.map do |params|
-        obsolete_enforced_style_message(params[:cop], params[:parameter],
-                                        params[:enforced_style],
-                                        params[:alternative])
+    # Default rules for obsoletions are in config/obsoletion.yml
+    # Additional rules files can be added with `RuboCop::ConfigObsoletion.files << filename`
+    def load_rules
+      rules = self.class.files.each_with_object({}) do |filename, hash|
```

Comment:
Just in case if some extension decides to override RuboCop's own cops, maybe `.reverse`? Or mix in `DEFAULT_RULE_FILE` right here, e.g. `[DEFAULT_RULE_FILE, *self.class.files].each...`?

Reply:
`self.class.files` is defaulted to `[DEFAULT_RULE_FILE]` with this in mind (so that `files << ...` will append but the default will continue to be first.

I'm not sure that an extension _should_ override something from RuboCop, but if it does, it probably should be applied?